app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,121 @@
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    TextField(
+                        value = number,
+                        onValueChange = { number = it },
+                        label = { Text(add_credit_card_number) },
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(horizontal = dimen_16_dp),
+                        colors = TextFieldDefaults.colors(
+                            focusedContainerColor = Color.White,
+                            unfocusedContainerColor = Color.White
+                        ),
+                    )
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(dimen_16_dp),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        TextField(
+                            modifier = Modifier.weight(1f),
+                            value = expirationMonth,
+                            onValueChange = { expirationMonth = it },
+                            label = { Text(add_credit_card_month) },
+                            colors = TextFieldDefaults.colors(
+                                focusedContainerColor = Color.White,
+                                unfocusedContainerColor = Color.White
+                            ),
+                        )
+                        Spacer(Modifier.width(dimen_16_dp))
+                        TextField(
+                            modifier = Modifier.weight(1f),
+                            value = expirationYear,
+                            onValueChange = { expirationYear = it },
+                            label = { Text(add_credit_card_year) },
+                            colors = TextFieldDefaults.colors(
+                                focusedContainerColor = Color.White,
+                                unfocusedContainerColor = Color.White
+                            ),
+                        )
+                    }
+
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(dimen_16_dp),
+                        horizontalArrangement = Arrangement.End,
+                    ) {
+                        TextButton(onClick = onDismissRequest) {
+                            Text(add_credit_card_cancel)
+                        }
+                        TextButton(onClick = onDismissRequest) {
+                            Text(add_credit_card_save)
```

Comment:
If I want to send the user's input back to the previous screen, what's the most efficient method? Should I create a data class and return its instance, or is it better to send the data back string by string?

Reply:
the best way is using a database in this case , which means you have access to your information always inside your app, but for now, if you want just to transfer your input data into another screen,the answer is yes , 
using a data class to encapsulate the user's input is generally a cleaner and more maintainable approach

Reply:
here is example of the dataclass that you can use :

```
data class CreditCardInfo(
    val name: String,
    val number: String,
    val expirationMonth: String,
    val expirationYear: String
)
```